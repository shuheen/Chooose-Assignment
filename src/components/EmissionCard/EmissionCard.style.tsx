import styled from '@emotion/styled';

const StyledEmissionCard = styled.div`
  width: 80%;
`;

export default StyledEmissionCard;
