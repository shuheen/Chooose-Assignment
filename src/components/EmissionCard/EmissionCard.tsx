import {Card, Flex, Text, Spacer} from '@chakra-ui/react';
import React from 'react';
import StyledEmissionCard from './EmissionCard.style';

interface EmissionCardProps {
  emission_offset: string;
}
function EmissionCard({emission_offset}: EmissionCardProps) {
  return (
    <StyledEmissionCard data-testid="emission-card">
      <Card backgroundColor="#1f2837" color="#fff" borderRadius="md" p={2} mt={2}>
        <Flex>
          <Text fontSize="12px">Emission</Text>
          <Spacer />
          <Text fontSize="12px" data-testid="emission-value">
            {emission_offset}
          </Text>
        </Flex>
      </Card>
    </StyledEmissionCard>
  );
}

export default EmissionCard;
