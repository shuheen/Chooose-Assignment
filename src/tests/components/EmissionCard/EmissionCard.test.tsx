// React
import {render, screen} from '@testing-library/react';

// Component
import EmissionCard from '../../../components/EmissionCard/EmissionCard';

//Defaults
const defaultProps = {
  emission_offset: '810 Kg CO2e',
};
// Tests
describe('Tests for <EmissionCard />', () => {
  it('Renders with defaults', () => {
    const {container} = render(<EmissionCard {...defaultProps} />);
    // Create the snapshot test
    expect(container).toMatchSnapshot();
  });

  it('should be render with emission offset value', () => {
    render(<EmissionCard {...defaultProps} />);
    const emissionValue = screen.getByTestId('emission-value');

    expect(emissionValue).toBeInTheDocument();
    expect(emissionValue).toHaveTextContent(defaultProps.emission_offset);
  });
});
