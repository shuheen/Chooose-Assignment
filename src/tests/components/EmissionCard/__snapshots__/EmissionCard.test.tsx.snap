// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <EmissionCard /> Renders with defaults 1`] = `
<div>
  <div
    class="css-1vgx9f6"
    data-testid="emission-card"
  >
    <div
      class="chakra-card css-ypyvwe"
    >
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-1qxtz39"
        >
          Emission
        </p>
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-1qxtz39"
          data-testid="emission-value"
        >
          810 Kg CO2e
        </p>
      </div>
    </div>
  </div>
</div>
`;
